### ISSUE TITLE
82580 L3 Timestamping

### ISSUE THREAD

**Issue by NetFalcon** [emmericp/MoonGen#181]
Hi there, 
I am very convinced that the 82580 L3 timestamping should work :) 
Is my assumption wrong ? Any suggestion on how to make it work ?

`[INFO]  Found 5 usable devices:
   Device 0: 64:00:6A:7B:35:C1 (Intel Corporation Ethernet Connection I217-LM)
   Device 1: A0:36:9F:40:58:0C (Intel Corporation I350 Gigabit Network Connection)
   Device 2: A0:36:9F:40:58:0D (Intel Corporation I350 Gigabit Network Connection)
   Device 3: 90:E2:BA:AD:38:0E (Intel Corporation 82580 Gigabit Network Connection)
   Device 4: 90:E2:BA:AD:38:0F (Intel Corporation 82580 Gigabit Network Connection)
[INFO]  Waiting for devices to come up...
[INFO]  Device 4 (90:E2:BA:AD:38:0F) is up: 1000 MBit/s
[INFO]  Device 3 (90:E2:BA:AD:38:0E) is up: 1000 MBit/s
[INFO]  2 devices are up.
[INFO]  Testing timestamping L2 PTP packets without rx filtering for 5 seconds.
Samples: 32723, Average: 350.4 ns, StdDev: 4.8 ns, Quartiles: 352.0/352.0/352.0 ns

[INFO]  Testing timestamping L2 PTP packets with rx filtering for 5 seconds.
Samples: 33021, Average: 350.4 ns, StdDev: 4.8 ns, Quartiles: 352.0/352.0/352.0 ns

[INFO]  Testing timestamping L2 PTP packets without rx filtering for 5 seconds.
[INFO]  Adding VLAN tag, this is not supported on some NICs.
Samples: 32899, Average: 350.4 ns, StdDev: 4.7 ns, Quartiles: 352.0/352.0/352.0 ns

[INFO]  Testing timestamping UDP packets to port 319 without rx filtering for 5 seconds.
Samples: 0, Average: nan ns, StdDev: 0.0 ns, Quartiles: nan/nan/nan ns
[ERROR] Received no packets.

[INFO]  Testing timestamping UDP packets to port 1234 without rx filtering for 5 seconds.
[WARN]  Timestamping on UDP ports other than 319 is not supported by the hardware or driver
Samples: 0, Average: nan ns, StdDev: 0.0 ns, Quartiles: nan/nan/nan ns
[ERROR] Received no packets.

[INFO]  Testing timestamping UDP packets to port 319 without rx filtering for 5 seconds.
[INFO]  Adding VLAN tag, this is not supported on some NICs.
Samples: 0, Average: nan ns, StdDev: 0.0 ns, Quartiles: nan/nan/nan ns
[ERROR] Received no packets.

[INFO]  Testing timestamping UDP packets to port 319 with rx filtering for 5 seconds.
Samples: 0, Average: nan ns, StdDev: 0.0 ns, Quartiles: nan/nan/nan ns
[ERROR] Received no packets.

[INFO]  Testing timestamping UDP packets to port 319 with rx filtering for 5 seconds.
[INFO]  Using multiple flows, this can be slower on some NICs.
Samples: 0, Average: nan ns, StdDev: 0.0 ns, Quartiles: nan/nan/nan ns
[ERROR] Received no packets.

[INFO]  Testing timestamping UDP packets to port 319 with rx filtering for 5 seconds.
[INFO]  Flooding link with UDP packets with the same flow 5-tuple.
[INFO]  This tests whether the filter matches on payload.
Samples: 0, Average: nan ns, StdDev: 0.0 ns, Quartiles: nan/nan/nan ns
[ERROR] Received no packets.
`

**Comment by emmericp** [comment]
Yes, that used to work before we used the DPDK driver to enable it. I'll look into it.

If just tested it with DPDK 17.05 and even the L2 timestamping fails on that branch...

**Comment by yyu4** [comment]
Hi Paul,

I'd like to know if you have solved this problem, since recently I tested 82580 NIC but failed to run L3 timestamping (by test-timestamping-capabilities.lua).

Thanks,
-Yi

> 
> 
> Yes, that used to work before we used the DPDK driver to enable it. I'll look into it.
> 
> If just tested it with DPDK 17.05 and even the L2 timestamping fails on that branch...
